src/pip/_internal/network/auth.py

```diff
@@ -109,7 +109,7 @@ def _get_index_url(self, url: str) -> Optional[str]:
     def _get_new_credentials(
         self,
         original_url: str,
-        allow_netrc: bool = True,
+        allow_netrc: bool = False,
         allow_keyring: bool = False,
```

Comment:
Since this is private interface anyway, let’s change these to required keyword-only arguments instead to make things more explicit.